app/models/goal.py

```diff
@@ -1,5 +1,22 @@
 from app import db
 
-
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    goal_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    tasks = db.relationship('Task', backref='goal', lazy=True)
+
+    def to_dict(self):
+        return {"id": self.goal_id,
+                "title": self.title,} 
+
+    @classmethod
+    def from_dict(cls, task_data):
+        new_goal = cls(title=task_data["title"])
+        return new_goal
+    
+    def tasks_to_dict(self):
+        return {
+            "id": int(self.goal_id),
+            "title": self.title,
+            "tasks": [task.to_dict() for task in self.tasks]
+        }
```

Comment:
I think we could combine this helper and the `to_dict` above and DRY out the code here a little bit! Remember that the pipe operator `|` can merge two dictionaries together, so we could shorten the above to:

```
def to_dict: 
    base_dict = {"id": self.goal_id, "title": self.title}

    if self.tasks: 
        tasks_dict = {"tasks": [task.to_dict() for task in self.tasks]}
        return base_dict | tasks_dict
    else:
        return base_dict
```